### PR TITLE
refactor(app): add robot cal check intro and clear deck warning

### LINKED ISSUE #5031: Deck Calibration Check: Clear Deck
As a user, I would like to be guided through clearing my deck, so that I have space for tipracks and access to the deck calibration crosses.

## Acceptance Criteria
- [x] After reviewing tip rack requirements, user views a modal instructing them to remove all modules and labware from the deck
- [x] User has option to **cancel** and exit to robots page
- [x] User has option to **continue** and proceed to 'Deck Calibration Check Complete' modal

## Copy
- See design

## Design
- https://zpl.io/bzejKW7

### LINKED ISSUE #5032: Deck Calibration Check: Intro and Tiprack Requirements
As an OT-2 user testing my robot’s calibration, I would like to know what to expect and what tip racks I need in advance, so that I have them on hand.

## Acceptance Criteria
- [x] On initiating calibration check, user views a modal informing them of what deck calibration does, what to expect, and what labware they will need
- [x] Modal informs user that the calibration check is intended to help them verify the accuracy of their OT-2's calibration
- [x] Modal informs user that they will require an Opentrons tiprack matching any mounted pipettes (one per unique tip type)
- [x] User informed that the use of Opentrons brand tips is critical for an effective test
- [x] Tip rack brand, model and image displayed in modal
- [x] User can **cancel** and exit flow to Robot page
- [x] Use can click **continue** and proceed to 'Deck Calibration Check Complete' modal

## Copy
- See design

## Design
- https://zpl.io/b6ExLKg

### PR DESCRIPTION
**Pull request by b-cooper**
## overview

Add the first steps of the robot calibration check flow to the run app, including the introduction
to required labware and warning that the deck must be empty to start.

**Note**: This pr does not include changes to the robot calibration check session accession. Following PR's will address the need logic surrounding existing robot calibration check sessions on the app and the robot.   

Closes #5031, Closes #5032

## changelog

- added `Introduction` component which is the corresponding view layer for the 'sessionStart' robot calibration check session step
- added functional unit tests for the `CheckCalibration` parent component, and the new `Introduction` component
- update api types to match api side changes

## review requests

- [ ] Start a Robot Calibration Check Session from the Robot Settings page. 
- [ ] After the session loads, you should first be brought to the new Introduction page and asked to prep the correct tiprack(s) corresponding to your attached pipettes.  (e.g. (P20S, P300M) -> (opentrons-20µl, opentrons-300µl) or (P1000S, P1000S) -> (opentrons-1000µl))
- [ ] Clicking continue from the introduction should lead too a warning modal telling you to clear your deck to continue.
- [ ] said warning modal should exit the flow if you click "Cancel", otherwise clicking "Continue" should log a debug statement for now. 

## risk assessment

low risk, all behind feature flag

### CODE REVIEW

File: app/src/calibration/constants.js [hunk 1]
```diff
@@ -21,5 +22,31 @@ export const DELETE_ROBOT_CALIBRATION_CHECK_SESSION_FAILURE: 'calibration:DELETE
...
+const PICK_UP_TIP: 'pickUpTip' = 'pickUpTip'
+const CHECK_POINT_ONE: 'checkPointOne' = 'checkPointOne'
+const CHECK_POINT_TWO: 'checkPointTwo' = 'checkPointTwo'
+const CHECK_POINT_THREE: 'checkPointThree' = 'checkPointThree'
+const CHECK_HEIGHT: 'checkHeight' = 'checkHeight'
+const SESSION_EXIT: 'sessionExit' = 'sessionExit'
+const BAD_ROBOT_CALIBRATION: 'badRobotCalibration' = 'badRobotCalibration'
+const NO_PIPETTES_ATTACHED: 'noPipettesAttached' = 'noPipettesAttached'
+
+export const ROBOT_CALIBRATION_CHECK_STEPS = {
```
Comment (mcous) [inline app/src/calibration/constants.js:41]: Nitpick, but I don't love this namespace object. I think it's a little easier to work with if we export the constants directly. If you're concerned about collisions, I would prefix the constant names, e.g. `CHECK_STEP_SESSION_START`

If you'd still like to export the namespace object, could we avoid all-caps in the name in favor of camel or pascal (my preference for namespaces) case?

File: app/src/calibration/api-types.js [hunk 2]
```diff
@@ -1,23 +1,55 @@
...
+}
 
 export type RobotCalibrationCheckSessionData = {|
-  instruments: { [string]: string },
+  instruments: { [string]: RobotCalibrationCheckInstrument },
   currentStep: RobotCalibrationCheckStep,
   nextSteps: {
     links: { [RobotCalibrationCheckStep]: string },
   },
+  labware: Array<RobotCalibrationCheckLabware>,
```
Comment (mcous) [inline app/src/calibration/api-types.js:54]: I'm not super familiar with what this array will look like coming from the API, but it's looking like we're throwing it right into the UI in the same order. This isn't a problem in-and-of itself, and it's something we do in a variety of places in the app, but we should keep an eye on the order stability if that's an actual concern

File: app/src/components/CheckCalibration/index.js [hunk 3]
```diff
@@ -24,33 +26,64 @@ type CheckCalibrationProps = {|
...
+    case ROBOT_CALIBRATION_CHECK_STEPS.CHECK_HEIGHT:
+    case ROBOT_CALIBRATION_CHECK_STEPS.SESSION_EXIT:
+    case ROBOT_CALIBRATION_CHECK_STEPS.BAD_ROBOT_CALIBRATION:
+    case ROBOT_CALIBRATION_CHECK_STEPS.NO_PIPETTES_ATTACHED: {
+      stepContents = <CompleteConfirmation robotName={robotName} exit={exit} />
+      modalContentsClassName = styles.terminal_modal_contents
+      break
+    }
+    default: {
+      stepContents = null
```
Comment (mcous) [inline app/src/components/CheckCalibration/index.js:74]: I'm testing on a Pi without pipettes attached, but from what I can tell, we hit this default in two cases that we're going to need to make sure we cover:

- Create request in flight
- Create request errors

**Review by mcous — APPROVED** [review]
I can't get very far in smoke testing this flow because create session 500s if there are no pipettes attached (aside: is 500 really the right error there? Maybe a 403 Forbidden?) and I don't have a full robot, but the code and unit tests look good!

Added a few comments for future work

**Review by sfoster1 — CHANGES_REQUESTED** [review]
image

different tiprack kinds have different spacing in the UI

The other stuff looks good though!

**Review by sfoster1 — APPROVED** [review]
All fixed! LGTM

**Pull request merged**